backend/api/Program.cs

```diff
@@ -1,16 +1,35 @@
 ﻿using Api.Configurations;
 using Api.Context;
 using Api.Services;
+using Azure.Identity;
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Azure;
 using Microsoft.Identity.Web;
 
 var builder = WebApplication.CreateBuilder(args);
 
+
+builder.Services.AddAzureClients(azureBuilder =>
+{
+    azureBuilder.AddSecretClient(builder.Configuration.GetSection("KeyVault"));
+
+    azureBuilder.UseCredential(new DefaultAzureCredential());
+});
+
 builder.Services.AddDbContext<FlotillaDbContext>(options => options.UseInMemoryDatabase("flotilla"));
+
+builder.Services.AddSingleton<DefaultAzureCredential>();
+
+builder.Services.AddSingleton<KeyVaultService>();
+
 builder.Services.AddScoped<RobotService>();
+
```

Comment:
No need for empty line between services we have made imo, but I like the empty line between different types of services